shader/svgf_atrous.comp

```diff
@@ -1,185 +1,140 @@
-layout(binding = 1, set = 0, rg16_snorm) uniform readonly image2DArray in_normal;
-layout(binding = 2, set = 0, rgba16f) uniform readonly image2DArray in_pos;
-layout(binding = 3, set = 0, rgba16f) uniform image2DArray inout_color[2];
-layout(binding = 4, set = 0, rgba16f) uniform readonly image2DArray in_albedo;
-layout(binding = 6, set = 0, rgba16f) uniform image2DArray inout_specular[2];
-layout(binding = 7, set = 0, rgba16f) uniform image2DArray moments_history[2];
-layout(binding = 8, set = 0, rgba16f) uniform image2DArray svgf_color_hist;
-layout(binding = 9, set = 0, rgba16f) uniform image2DArray svgf_hist_specular;
-
-layout(push_constant) uniform push_constant_buffer
+layout(binding = 0, set = 0, rgba32f) uniform image2DArray diffuse_ping;
+layout(binding = 1, set = 0, rgba32f) uniform image2DArray diffuse_pong;
+layout(binding = 2, set = 0, rgba32f) uniform image2DArray diffuse_hist;
+layout(binding = 3, set = 0, rgba32f) uniform image2DArray final_output;
+layout(binding = 4, set = 0, rgba32f) uniform image2DArray in_linear_depth;
+layout(binding = 5, set = 0, rgba32f) uniform image2DArray in_normal;
+layout(binding = 6, set = 0, rgba32f) uniform image2DArray in_albedo;
+layout(binding = 7, set = 0, rgba32f) uniform image2DArray specular_ping;
+layout(binding = 8, set = 0, rgba32f) uniform image2DArray specular_pong;
+layout(binding = 9, set = 0, rgba32f) uniform image2DArray spec_hist;
+layout(binding = 10, set = 0, rgba32f) uniform image2DArray in_material;
+layout(binding = 11, set = 0, rgba32f) uniform image2DArray in_moments;
+
+float filter_variance(ivec3 p)
 {
-    ivec2 size;
-    int parity;
-    int level;
-    int stride;
-    int iteration_count;
-} control;
-
-vec4 sample_color(ivec3 p)
-{
-    if(control.parity == -1) return imageLoad(in_color, p);
-    else return imageLoad(inout_color[control.parity], p);
-}
-
-vec4 sample_specular(ivec3 p)
-{
-    return imageLoad(inout_specular[(control.parity+1)&1], p);
-}
-
-void write_color(ivec3 p, vec4 color)
-{
-    imageStore(inout_color[(control.parity+1)&1], p, color);
-}
+    float sum = 0.f;
 
-void write_specular(ivec3 p, vec4 color)
-{
-    imageStore(inout_specular[control.parity&1], p, color);
-}
-
-float compute_sigma_luminance(ivec3 pos, float center)
-{
     const int r = 1;
-    float sum = center * gaussian_kernel[0][0];
-
     for (int yy = -r; yy <= r; ++yy)
     for (int xx = -r; xx <= r; ++xx)
     {
-        if (xx == 0 && yy == 0)
-            continue;
-
-        const ivec3 o = ivec3(
-            clamp(pos.xy + ivec2(xx, yy), ivec2(0), control.size-1),
-            pos.z
-        );
-        float variance = sample_color(pos).a;
-        float w = gaussian_kernel[abs(xx)][abs(yy)];
-        sum += variance * w;
+        const ivec3 q = p + ivec3(xx, yy, 0.0);
+        const float k = gaussian_kernel[abs(xx)][abs(yy)];
+        float v = (control.level & 1) == 0 ? imageLoad(diffuse_ping, q).a : imageLoad(diffuse_pong, q).a;
```

Comment:
Consider swapping diffuse_ping and diffuse_pong via push descriptors to clean up this selection of the source (and same elsewhere in this same file, too).